skyportal/facility_apis/sedmv2.py

```diff
@@ -0,0 +1,233 @@
+from . import FollowUpAPI
+from baselayer.app.env import load_env
+from datetime import datetime, timedelta
+import json
+import requests
+
+from ..utils import http
+
+env, cfg = load_env()
+
+
+def validate_request_to_sedmv2(request):
+    """Validate FollowupRequest contents for SEDMv2 queue.
+
+    Parameters
+    ----------
+    request: skyportal.models.FollowupRequest
+        The request to send to SEDM.
+
+    method_value: 'new', 'edit', 'delete'
+        The desired SEDMv2 queue action.
+    """
+
+    for param in [
+        "observation_choices",
+        "exposure_time",
+        "exposure_counts",
+        "maximum_airmass",
+        "minimum_lunar_distance",
+        "priority",
+        "start_date",
+        "end_date",
+    ]:
+        if param not in request.payload:
+            raise ValueError(f'Parameter {param} required.')
+
+    for filt in request.payload["observation_choices"]:
+        if filt not in ["IFU", "g", "r", "i", "z"]:
+            raise ValueError(f'Filter configuration {filt} unknown.')
+
+
+class SEDMV2API(FollowUpAPI):
+    """SkyPortal interface to the Spectral Energy Distribution machine (SEDMv2)."""
+
+    @staticmethod
+    def submit(request):
+        """Submit a follow-up request to SEDMv2.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to submit.
+        """
+
+        from ..models import FacilityTransaction, DBSession
+
+        validate_request_to_sedmv2(request)
```

Comment:
If this raises an error (ie. ill-formed request) then the error will not handled properly. Need to wrap this in a try/except statement and `return self.error(...)` something sensible if an error is raised.

Reply:
All of the other API's just raise Errors. What class is taking care of the error handling here?

Reply:
`validate_request_to_sedmv2` either raises an error (`ValueError`) or returns None. Raising this error within an API request will not necessarily return back the sender a response they can parse. [`baselayer` overwrites the tornado `error` method](https://github.com/cesium-ml/baselayer/blob/ec952bbb3fcde91c19a9c2397c38f16e3d6b29c7/app/handlers/base.py#L286) to handle this gracefully.  So I'd do something like:

```
        try:
            validate_request_to_sedmv2(request)
        except (ValueError, ) as e:
            return self.error('SEMDv2 validation error: {e.normalized_messages()}')
```

Reply:
ok but that self isn't available anywhere here is it? this is a static method called in handlers/api/followup_request.py. The exception ends up in the red bar text on the front end.

Reply:
Ah, you're right. I thought that these APIs were subclasses of the API handlers we have in the rest of the app. But instead it's a base of `FollowUpAPI`. So what `self.` is indeed available here `self.error` has not been implemented. I'd argue we should implement it in the `FollowUpAPI` (or its base class) but that's beyond the scope of this PR.